R/bed_makewindows.r

```diff
@@ -52,13 +52,16 @@ bed_makewindows <- function(x, genome, win_size = 0,
   if (win_size == 0 && num_win == 0)
     stop('specify either `win_size` or `num_win`', call. = FALSE)
   
-  res <- purrr::by_row(x, split_interval, genome, win_size,
-           step_size, num_win, 
-           reverse, .collate = 'rows',
-           .labels = FALSE)
-  
-  res <- select(res, -.row)
-    
+  x <- rowwise(x)
+  #pass rows to split_intervals, by default do() will convert to list
+  res <- do(x, 
+            split_interval(as_data_frame(.),
```

Comment:
Is the `as_data_frame` necessary here (is this what the "covert to list" comment is aboue)? I think this is the thing causing you to add `.` to globals, which is weird and I think could cause problems down the line.